front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +51,51 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points!;
```

Comment:
Are you sure that `operational_points` can't be null or undefined ?

Reply:
Yep. As long as we ask for a data (ops, curves...), the endpoint will return them (the pathfinding has already been done with the previous endpoint : pathfindingblocks)